pkg/controller/nodes/dynamic/dynamic_workflow.go

```diff
@@ -321,6 +323,16 @@ func (d dynamicNodeTaskNodeHandler) progressDynamicWorkflow(ctx context.Context,
 					nil
 			}
 			o = &handler.OutputInfo{OutputURI: destinationPath}
+			if d.eventConfig.RawOutputPolicy == config.Inline {
+				var outputData = &core.LiteralMap{}
+				err := nCtx.DataStore().ReadProtobuf(ctx, destinationPath, outputData)
```

Comment:
I wonder if it's a bit better to to construct an OutputWriter instead using the outputUrl but not really reading the outputs right here... just delaying that till a later point (maybe right before sending the event?)